go-controller/pkg/ovn/service.go

```diff
@@ -57,14 +71,107 @@ func (ovn *Controller) syncServices(services []interface{}) {
 			if util.ServiceTypeHasNodePort(service) {
 				port := fmt.Sprintf("%d", svcPort.NodePort)
 				nodeportServices[svcPort.Protocol] = append(nodeportServices[svcPort.Protocol], port)
+				gatewayRouters, _, err := ovn.getOvnGateways()
+				if err == nil {
+					for _, gatewayRouter := range gatewayRouters {
+						lb, err := ovn.getGatewayLoadBalancer(gatewayRouter, svcPort.Protocol)
+						if err != nil {
+							klog.Warningf("Service Sync: Gateway router %s does not have load balancer (%v)",
+								gatewayRouter, err)
+							continue
+						}
+						physicalIPs, err := ovn.getGatewayPhysicalIPs(gatewayRouter)
+						for _, physicalIP := range physicalIPs {
+							if err != nil {
+								klog.Warningf("Service Sync: Gateway router %s does not have physical ip (%v)",
+									gatewayRouter, err)
+								continue
+							}
+							name := ovn.generateACLName(lb, physicalIP, svcPort.NodePort)
```

Comment:
L85: Not sure why you are checking for `err` inside the for loop here. I initially thought that maybe you wanted to print a warning for every physical IP, but the error message is not printing the physicalIP. So, should you move this check outside of the for loop?

Reply:
copy paste error from when I was copying this logic from createService. Thanks for catching it. I'll post a fix shortly.